lib/cublas/linalg.jl

```diff
@@ -39,7 +39,7 @@ function LinearAlgebra.dot(x::StridedCuArray{T1}, y::StridedCuArray{T2}) where {
         start = (blockIdx().x - 1) * block_stride + 1
         stop = blockIdx().x * block_stride
 
-        cache = @cuStaticSharedMem(T, (512#= MAX_THREADS =#,))
+        cache = @cuStaticSharedMem(T, (256#= MAX_THREADS =#,))
```

Comment:
Why not a dynamic cache?

```
cache = CuDynamicSharedArray(T, (thread_stride,))
...
config = launch_configuration(kernel.fun; shmem=(threads)->threads*sizeof(T))
...
kernel(x, y, res, T; threads, blocks, shmem=threads*sizeof(T))
```

Also note the macros have been replaced by functions.

Reply:
Yes, the dynamic cache is just what I want here. Thanks!